specs/merge/beacon-chain.md

```diff
@@ -354,13 +354,18 @@ def process_execution_payload(state: BeaconState, payload: ExecutionPayload, exe
 ## Testing
 
 *Note*: The function `initialize_beacon_state_from_eth1` is modified for pure Merge testing only.
-
-*Note*: The function `initialize_beacon_state_from_eth1` is modified: (1) using `MERGE_FORK_VERSION` as the current fork version, (2) utilizing the Merge `BeaconBlockBody` when constructing the initial `latest_block_header`, and (3) initialize `latest_execution_payload_header`.
+Modifications include:
+1. Use `MERGE_FORK_VERSION` as the current fork version
+2. Utilize the Merge `BeaconBlockBody` when constructing the initial `latest_block_header`
+3. Initialize `latest_execution_payload_header`.
+  If `execution_payload_header == ExecutionPayloadHeader()`, then the Merge has not yet occurred.
+  Else, the Merge starts from genesis.
 
 ```python
 def initialize_beacon_state_from_eth1(eth1_block_hash: Bytes32,
                                       eth1_timestamp: uint64,
-                                      deposits: Sequence[Deposit]) -> BeaconState:
+                                      deposits: Sequence[Deposit],
+                                      execution_payload_header: ExecutionPayloadHeader) -> BeaconState:
```

Comment:
Do we want to have a default value for `execution_payload_header`? It would then be compatible with the previous versions of this function but could hide some bugs as if related function calls are not updated whenever they are supposed to be it would return the state after the merge fork and before the transition

Reply:
`execution_payload_header: ExecutionPayloadHeader=ExecutionPayloadHeader()` probably could fix the tests immediately:

```python
def initialize_beacon_state_from_eth1(
        eth1_block_hash: Bytes32,
        eth1_timestamp: uint64,
        deposits: Sequence[Deposit],
        execution_payload_header: ExecutionPayloadHeader=ExecutionPayloadHeader()) -> BeaconState:
```